### ISSUE TITLE
Regression no-catch-shadow (1.1.0)

### ISSUE THREAD

**Issue by spadgos** [eslint/eslint#3322]
It looks like 1.1.0 introduced a regression in the `no-catch-shadow` rule: it now complains about every `catch`.

```js
function foo() {
  try {
    bar();
  } catch (e) { }
}
```
"Value of 'e' may be overwritten in IE 8 and earlier  no-catch-shadow"

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by sublimator** [comment]
Confirmed, we are experiencing this issue too @ [ripple-lib](https://github.com/ripple/ripple-lib)

version:
`v1.1.0`

source:
```js
'use strict';

function broken() {
  try {
    throw new Error();
  } catch (e) {
    //
  }
}

module.exports = broken;
```

config:
[eslintrc](https://raw.githubusercontent.com/ripple/javascript-style-guide/3cf3cf978839d827218654a2b29462af8309319d/eslintrc)

output:
```
src/broken.js
  6:5  error  Value of 'e' may be overwritten in IE 8 and earlier  no-catch-shadow
```

**Comment by BYK** [comment]
I think this is related to `eslint-babel` since I cannot reproduce with `espree`.

**Comment by sublimator** [comment]
Ah, thanks!

**Comment by spadgos** [comment]
So what's the workaround, exactly?

**Comment by sublimator** [comment]
what @spadgos said :)

**Comment by BYK** [comment]
@spadgos, @sublimator - I installed the latest version, which is `4.0.5`, and couldn't reproduce the issue. I looked at https://github.com/ripple/ripple-lib/blob/develop/package.json#L36 to see ripple-lib still at 3.x branch so I think that's the issue. If you upgrade it to 4.x I expect the issue to go away.

**Comment by sublimator** [comment]
@BYK 

Thanks! I should have checked that! I just updated another repo to the `@latest` babel-* and was confused. I'll give that a try.

**Comment by sublimator** [comment]
Updating did not work :( 

I've narrowed it down to this conf
```
rules:
  no-catch-shadow: 2
ecmaFeatures:
  blockBindings: true
```

By changing blockBindings to false, it stops emitting the error

**Comment by sublimator** [comment]
Also, downgrading to eslint 1.0 and error will disappear also. So it seems to me a regression in eslint 1.1?

**Comment by sublimator** [comment]
Can anyone else repro with that reduced conf and npm eslint 1.1.0 ? Any ideas?

**Comment by BYK** [comment]
@sublimator - Great work narrowing it down. I can now reproduce! Investigating the root cause.

**Comment by BYK** [comment]
@sublimator - Submitted #3360

**Comment by sublimator** [comment]
Awesome :)
